app/main.py

```diff
@@ -1,3 +1,62 @@
+from __future__ import annotations
+
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: Union[int, float]) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            return Distance(self.km + other.km)
+        return Distance(self.km + other)
+
+    def __iadd__(self, other: Union[int, float, Distance]) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+            return self
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: Union[int, float, Distance]) -> Distance:
```

Comment:
`other` can't be `Distance` here, only `int` or `float`, and you don't need to check its type

Reply:
and remove `Distance` from annotation here

Reply:
> and remove `Distance` from annotation here

changed